core/common-util/src/main/java/uk/gov/gchq/gaffer/commonutil/iterable/LimitedCloseableIterator.java

```diff
@@ -52,12 +64,18 @@ public void close() {
 
     @Override
     public boolean hasNext() {
-        boolean hasNext = (null == end || index < end) && iterator.hasNext();
-        if (!hasNext) {
+        boolean hasNext = (null == end || index < end);
+        boolean iteratorHasNext = iterator.hasNext();
```

Comment:
If hasNext is false then we don't actually need to call iterator.hasNext(). Calling iterator.hasNext() could mean loading the next batch from accumulo and could be expensive.